mirgecom/profiling.py

```diff
@@ -94,6 +95,11 @@ def _finish_profile_events(self) -> None:
 
         self.profile_events = []
 
+    def clear_profiling_data(self) -> None:
```

Comment:
This is an inherently stateful API, which I don't like. Maybe add the notion of a `ProfilingDataContainer` and allow the user to set that in the array context.

Reply:
Ok, modified the PR to just fix the footprint warning and will that the data container in a separate PR.